### ISSUE TITLE
Circuit Breaking doesn't work with TLS Origination

### ISSUE THREAD

**Issue by hubermark** [istio/istio#19262]
**Bug description**
I followed [Circuit Breakin](https://istio.io/docs/tasks/traffic-management/circuit-breaking/) instructions and adapted it to our system but the circuit breaker configuration is ignored when using TLS Origination

**Expected behavior**
Circuit Breacing config works with TLS Origination same as without (HTTP standard)

**Steps to reproduce the bug**
Set outbound traffic policy mode to REGISTRY_ONLY

1. check if an outgoing call is now forbidden for not configured hosts (out of pod)
``curl -I http://httpstat.us/200?sleep=100` -> HTTP/1.1 502 Bad Gateway

2. Allow traffic for httpstat.us, create VirtualService with timeout and DestinationRule with Circuit Breaking config. Working example without TLS Origination 
```
kubectl apply -f - <<EOF
---
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
    name: se-httpstat
spec:
    hosts:
        - httpstat.us
    ports:
        - number: 80
          name: http
          protocol: HTTP
    resolution: DNS
    location: MESH_EXTERNAL
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
    name: vs-httpstat
spec:
    hosts:
        - httpstat.us
    http:
        - name: "httpstat-route"
          route:
              - destination:
                    host: httpstat.us
                    subset: httpstat-origination
          timeout: 0.8s
---
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: dr-httpstat
spec:
  host: httpstat.us
  subsets:
    - name: httpstat-origination
  trafficPolicy:
    outlierDetection:
      consecutiveErrors: 1
      interval: 1s
      baseEjectionTime: 30s
      maxEjectionPercent: 100
EOF
```
3. Check if an outgoing call is now allowed and works within timeout boundary (out of pod)
`curl -v http://httpstat.us/200?sleep=100` -> HTTP/1.1 200 OK

4. Check timeout and Circuit Breaking works
`curl -I http://httpstat.us/200?sleep=1000`
First trial ends with Timeout from envoy
```
HTTP/1.1 504 Gateway Timeout
content-length: 24
content-type: text/plain
...
server: envoy
```
Next trial returns HTTP 503 from envoy due to Circuit Breaking is working
```
HTTP/1.1 503 Service Unavailable
content-length: 19
content-type: text/plain
...
server: envoy
```

5. Adapt these configuration to TLS Origination use. First check outgoing call is forbidden for not configured hosts (out of pod)
`curl -I http://www.google.com` -> HTTP/1.1 502 Bad Gateway

6. Allow traffic for www.google.com, create VirtualService with timeout and DestinationRule with Circuit Breaking config and TLS Origination
```
kubectl apply -f - <<EOF
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: se-google
spec:
  hosts:
  - www.google.com
  ports:
  - number: 443
    name: https-google
    protocol: HTTPS
  resolution: DNS
  location: MESH_EXTERNAL
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: vs-google
spec:
  hosts:
  - www.google.com
  http:
  - match:
    - port: 80
    route:
    - destination:
        host: www.google.com
        port:
          number: 443
    timeout: 0.001s # Even to short for google
---
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: dr-google
spec:
  host: www.google.com
  trafficPolicy:
    outlierDetection:
      consecutiveErrors: 1
      interval: 1s
      baseEjectionTime: 30s
      maxEjectionPercent: 100
    portLevelSettings:
    - port:
        number: 443
      tls:
        mode: SIMPLE
EOF
```
7. Check if an outgoing call is now allowed and works within timeout (Set timeout to 0.5s) boundary (out of pod)
`curl -I http://www.google.com` -> HTTP/1.1 200 OK

8. Check Timeout and Circuit Breaking - Set timeout to 0.001s (out of pod)
`curl -I http://www.google.com`
First trial ends with Timeout from envoy - Timout is working correctly
```
HTTP/1.1 504 Gateway Timeout
content-length: 24
content-type: text/plain
...
server: envoy
```
Further trials also return only HTTP 504 instead of the expected HTTP 503

I used www.google.com as an example to hide the real HTTPS endpoint I want to call but the effect is the same.

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
Istio:
client version: 1.2.0
citadel version: 1.3.3
egressgateway version: 1.3.3
galley version: 1.3.3
ingressgateway version: 1.3.3
pilot version: 1.3.3
policy version: 1.3.3
sidecar-injector version: 1.3.3
telemetry version: 1.3.3

Kubernetes
Client Version: version.Info{Major:"1", Minor:"14",
Server Version: version.Info{Major:"1", Minor:"14+"

Helm
Client: &version.Version{SemVer:"v2.14.1"

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**
GCP

**Comment by howardjohn** [comment]
Is this the same as https://github.com/istio/istio/issues/19101?

**Comment by hubermark** [comment]
Hello @howardjohn,
both tickets are from me but I think it's quite a difference in it.
I think it could be same code area but the occurrence is a little different

**Comment by howardjohn** [comment]
@hubermark sorry not sure why I thought they were the same, I understand now.

I can reproduce this, thanks for the config. I will take a look

**Comment by howardjohn** [comment]
I think this is actually caused by two misconfigurations

```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: se-google
spec:
  hosts:
  - httpbin.org
  ports:
  - number: 443
    name: https-google
    protocol: HTTPS
  resolution: DNS
  location: MESH_EXTERNAL
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: vs-google
spec:
  hosts:
  - httpbin.org
  http:
  - match:
    - port: 80
    route:
    - destination:
        host: httpbin.org
        port:
          number: 443
---
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: dr-google
spec:
  host: httpbin.org
  trafficPolicy:
    portLevelSettings:
    - port:
        number: 443
      tls:
        mode: SIMPLE
      outlierDetection:
        consecutiveErrors: 1
        interval: 1s
        baseEjectionTime: 30s
        maxEjectionPercent: 100
```

Then send `curl httpbin.org/status/501` a few times and it will switch to 503.

The problem with httpstat is the subset doesn't actually select the service entry, so its empty, resulting in 503
The problem with the google.com one is the portLevel setting replaces, not merges, with the global settings. So you don't actually apply the outlier detection

This was pretty hard to find though, I am not sure how we can detect situations like this

**Comment by howardjohn** [comment]
I think this is user error that is fixed in Istio 1.6 by validating properly. Additional validation is tracked in https://github.com/istio/istio/issues/24860

**Comment by Deepak275** [comment]
> 

@howardjohn This was the exact issue with the indentation of portLevelSettings and outlierDetection. Surprisingly, it is not mentioned in anywhere in the docs, so after spending 2 days debugging, stumble upon your comments 😅 and saved the day 😄 . But it would be helpful if it can be added in the docs itself with example. Thanks for this awesome product 😄 

> I think this is actually caused by two misconfigurations
> 
> ```
> apiVersion: networking.istio.io/v1alpha3
> kind: ServiceEntry
> metadata:
>   name: se-google
> spec:
>   hosts:
>   - httpbin.org
>   ports:
>   - number: 443
>     name: https-google
>     protocol: HTTPS
>   resolution: DNS
>   location: MESH_EXTERNAL
> ---
> apiVersion: networking.istio.io/v1alpha3
> kind: VirtualService
> metadata:
>   name: vs-google
> spec:
>   hosts:
>   - httpbin.org
>   http:
>   - match:
>     - port: 80
>     route:
>     - destination:
>         host: httpbin.org
>         port:
>           number: 443
> ---
> apiVersion: networking.istio.io/v1alpha3
> kind: DestinationRule
> metadata:
>   name: dr-google
> spec:
>   host: httpbin.org
>   trafficPolicy:
>     portLevelSettings:
>     - port:
>         number: 443
>       tls:
>         mode: SIMPLE
>       outlierDetection:
>         consecutiveErrors: 1
>         interval: 1s
>         baseEjectionTime: 30s
>         maxEjectionPercent: 100
> ```
> 
> Then send `curl httpbin.org/status/501` a few times and it will switch to 503.
> 
> The problem with httpstat is the subset doesn't actually select the service entry, so its empty, resulting in 503 The problem with the google.com one is the portLevel setting replaces, not merges, with the global settings. So you don't actually apply the outlier detection
> 
> This was pretty hard to find though, I am not sure how we can detect situations like this
